src/lando/api/legacy/workers/uplift_worker.py

```diff
@@ -0,0 +1,238 @@
+            self.apply_patch(repo, job, scm, uplift_revision)
+
+        # On success: create patches.
+        response = self.moz_phab_uplift(
+            job, user.profile.phabricator_api_key, base_revision
+        )
+
+        # Retrieve created revision IDs and tip revision ID.
+        commits = response["commits"]
+        created_revision_ids = [int(commit["rev_id"]) for commit in commits]
+        tip_revision_id = created_revision_ids[-1]
+
+        _, created = UpliftRevision.objects.get_or_create(
+            revision_id=tip_revision_id,
+            defaults={"assessment": multi_request.assessment},
+        )
+        if not created:
+            UpliftRevision.objects.filter(revision_id=tip_revision_id).update(
+                assessment=multi_request.assessment
+            )
+
+        # Trigger a Celery task to update the form on Phabricator.
+        self.call_task(
+            set_uplift_request_form_on_revision,
+            tip_revision_id,
+            multi_request.assessment.to_conduit_json_str(),
+            user.id,
+        )
+
+        job.created_revision_ids = created_revision_ids
+        job.transition_status(JobAction.SUCCESS)
+
+        return created_revision_ids
+
+    def notify_success(
+        self,
+        repo_label: str,
+        job_url: str,
+        recipient_email: str,
+        created_revision_ids: list[int],
+    ) -> None:
+        if not recipient_email:
+            return
+        revision_labels = [f"D{rev_id}" for rev_id in created_revision_ids]
+        self.call_task(
+            send_uplift_success_email,
+            recipient_email,
+            repo_label,
+            job_url,
+            revision_labels,
+        )
+
+    def notify_uplift_failure(
+        self,
+        job: UpliftJob,
+        repo_label: str,
+        job_url: str,
+        recipient_email: str,
+        reason: str,
+    ) -> None:
+        if not recipient_email:
+            return
+
+        has_conflicts = bool(job.error_breakdown)
+        conflict_sections = self.conflict_sections(job) if has_conflicts else None
+        failure_reason = job.error or reason
+        self.call_task(
+            send_uplift_failure_email,
+            recipient_email,
+            repo_label,
+            job_url,
+            failure_reason,
+            conflict_sections,
+        )
+
+    # TODO(sheehan): look at this
+    def conflict_sections(self, job: UpliftJob) -> list[dict[str, str]]:
+        breakdown = job.error_breakdown
+        if not breakdown:
+            return []
+
+        rejects = breakdown.get("rejects_paths") or {}
+        sections: list[dict[str, str]] = []
+        for path, data in rejects.items():
+            snippet = ""
+            content = data.get("content")
+            if content:
+                lines = content.splitlines()
+                snippet = "\n".join(lines[:20])
+            sections.append({"path": path, "snippet": snippet})
+        return sections
```

Comment:
We should already have some logic to do that in the SCM. Could we reuse/extend that? https://github.com/mozilla-conduit/lando/blob/c156d48d60870f7018436e19ad58aad6fb4f1db9/src/lando/main/scm/git.py#L287-L336

Maybe the “merge” there is a misnomer.

Reply:
After reading this through, I think it's simpler to just link to the job details page in the email with a note that merge conflicts are displayed there. Then I can re-use the merge conflict display from the landing jobs, and if folks want merge conflicts in the email I can add them at a later time. I'll fix this shortly.